app/scripts/modules/kubernetes/src/v2/pipelines/stages/patchManifest/PatchManifestStageForm.tsx

```diff
@@ -0,0 +1,174 @@
+    this.props.formik.setFieldValue(
+      'requiredArtifactIds',
+      bindings.filter(b => b.expectedArtifactId).map(b => b.expectedArtifactId),
+    );
+    this.props.formik.setFieldValue('requiredArtifacts', bindings.filter(b => b.artifact));
+  };
+
+  private handleRawManifestChange = (rawManifest: string, manifests: any): void => {
+    this.setState({
+      rawManifest,
+    });
+    this.props.formik.setFieldValue('patchBody', manifests[0]);
+  };
+
+  private onManifestSelectorChange = (): void => {};
+
+  private getSourceOptions = (): Array<Option<string>> => {
+    return map([this.textSource, this.artifactSource], option => ({
+      label: capitalize(option),
+      value: option,
+    }));
+  };
+
+  public render() {
+    const stage = this.props.formik.values;
+    return (
+      <div className="container-fluid form-horizontal">
+        <h4>Resource to Patch</h4>
+        <ManifestSelector
+          application={this.props.application}
+          modes={[SelectorMode.Static, SelectorMode.Dynamic]}
+          onChange={this.onManifestSelectorChange}
+          selector={stage as any}
+        />
+        <hr />
+        <h4>Patch Content</h4>
+        <StageConfigField label="Manifest Source" helpKey="kubernetes.manifest.source">
+          <RadioButtonInput
+            options={this.getSourceOptions()}
+            onChange={(e: any) => this.props.formik.setFieldValue('source', e.target.value)}
+            value={stage.source || 'text'}
+            inline={true}
+          />
+        </StageConfigField>
+        {stage.source === this.textSource && (
+          <StageConfigField label="Manifest">
+            <YamlEditor onChange={this.handleRawManifestChange} value={this.state.rawManifest} />
+          </StageConfigField>
+        )}
+        {stage.source === this.artifactSource && (
+          <>
+            <StageArtifactSelectorDelegate
+              artifact={stage.manifestArtifact}
+              excludedArtifactTypePatterns={this.excludedManifestArtifactTypes}
+              expectedArtifactId={stage.manifestArtifactId}
+              helpKey="kubernetes.manifest.expectedArtifact"
+              label="Manifest Artifact"
+              onArtifactEdited={this.onManifestArtifactEdited}
+              onExpectedArtifactSelected={(artifact: IExpectedArtifact) => this.onManifestArtifactSelected(artifact.id)}
+              pipeline={this.props.pipeline}
+              selectedArtifactAccount={stage.manifestArtifactAccount}
+              selectedArtifactId={stage.manifestArtifactId}
+              setArtifactAccount={this.onManifestArtifactAccountSelected}
+              setArtifactId={this.onManifestArtifactSelected}
+              stage={stage}
+              updatePipeline={this.props.updatePipeline}
+            />
+            <StageConfigField label="Expression Evaluation" helpKey="kubernetes.manifest.skipExpressionEvaluation">
```

Comment:
is this just copied from the deploy stage? we do not currently support skipping spel expression evaluation in the patch manifest stage

Reply:
it is. is it something we want to support? the corresponding changes in Orca that extract common logic for Patch and Deploy manifest now permits this.

Reply:
let's not expose it here, the user story we were trying to support with that change was a deploy stage following a helm bake stage. happy to be proven wrong in future but for now haven't heard anyone demand this in the patch stage so would rather not add the complexity

Reply:
sounds good. the code in orca defaults `skipExpressionEvaluation` to `false`, so by not including the option in the UI, it works as designed currently.